src/controllers/JobController.ts

```diff
@@ -77,4 +78,107 @@ export class JobController {
       count: jobs.length,
     });
   }
+
+  //Create a new job
+  async createJob(req: Request, res: Response): Promise<void> {
+    const {
+      jobRoleName,
+      description,
+      jobSpecLink,
+      responsibilities,
+      numberOfOpenPositions,
+      location,
+      closingDate,
+      band,
+      capability,
+    } = req.body;
+
+    if (
+      !jobRoleName ||
+      !description ||
+      !jobSpecLink ||
+      !responsibilities ||
+      !numberOfOpenPositions ||
+      !location ||
+      !closingDate ||
+      !band ||
+      !capability
+    ) {
+      throw new BusinessError("All fields are required to create a job", 400);
+    }
```

Comment:
it would be good to extract this to a reusable validator class (and add tests to it :)
Then inject it to the controller.